department-of-veterans-affairs/vets-website
# Pensions 76767 update marriage history

## Summary
Update marriage history pages to use web component patterns where applicable on the Pension Benefits form (21P-527EZ)

### Note
The 'currentOrPastDate' web component pattern does not function correctly when implemented within a 'pagePerItem' array list loop item. An issue will be opened with platform to fix.

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`

## How to verify
1. Go to `http://localhost:3001/pension/application/527EZ` in your browser
2. Go to the 'Marital status' page in the 'Household information' chapter
3. Select 'Married'
4. Enter at least 2 marriages
5. Verify that a previous marriage is title'd correctly and components are updated
6. Verify that the previous marriage is validated
7. Verify that current marriage is title'd correctly and components are updated
8. Verify that the current marriage is validated
9. Verify that the data is saved to the api and redux
10. Continue to the 'Review application' page
11. Verify that the data is viewable and editable
12. Verify that the application is able to be submitted

## Acceptance criteria
[Issue in Github - 76767](https://github.com/department-of-veterans-affairs/va.gov-team/issues/76767)

## Related issue(s)
n/a

## Testing done

## Screenshots
### Previous Marriage
| Before | After |
| ------ | ------ |
|![localhost_3001_pension_application_527EZ_form-saved (20)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/fb49d2c2-1f55-4f0c-9735-ae4e2148c455)|![localhost_3001_pension_application_527EZ_form-saved (18)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/0cc9f3e5-ca47-425c-8eda-18754a59d2af)

### Current Marriage
| Before | After |
| ------ | ------ |
|![localhost_3001_pension_application_527EZ_form-saved (19)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/39fadefc-60ab-414c-a3da-f22746d7889a)|![localhost_3001_pension_application_527EZ_form-saved (3)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/04615d2f-1215-4b85-952f-694ceeeeaec0)


## What areas of the site does it impact?
Pension Benefits Application

### Quality Assurance & Testing

### Error Handling

### Authentication
